### PR TITLE
Pagination issues

### PR DESCRIPTION
**Pull request by emma-steuer**

### CODE REVIEW

**Comment by sgillies** [comment]
@joferkington as the reporter of the issue and the one with the most basemaps experience, can you review this?

File: jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb [hunk 1]
```diff
@@ -343,72 +420,164 @@
...
    "metadata": {},
    "source": [
-    "#### Reproject\n",
-    "\n",
-    "Reprojecting a basemap is necessary when the original map projection does not accurately represent the Earth's surface in a specific region or area. This can happen when the imagery for the basemap comes from a place far from the equator. In the example shown in this notebook, there is not a significant change in circumference of the earth from the equator, but it is always best to check using the code below. \n",
-    "\n",
-    "Find the latitude of the area of interest and input it as a parmameter. The code below finds the correct resolution for the area of interest."
+    "### Batch quad IDs and place the orders\n",
+    "---\n",
+    "It is only possible to place orders with 100 quad IDs at a time. This function will batch the quad IDs into lists of 100 each and place each of those orders. "
```
Comment (joferkington) [inline jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb:467]: Isn't it 50 if merge is used?  Or did that get increased to 100?

**Review by joferkington — APPROVED** [review]
Thanks for this, and it's a definite improvement!!  The pagination is quite important and this fixes the issue there, as well as being an improvement overall.

My one comment is that we're showing the polygon search in the orders API sections and not the general basemaps API sections.  I wonder if it might make sense to demonstrate ordering a basemap by AOI directly in the orders section and have the polygon search portions in the basemaps API demos?

If you're using the orders API, at least for small AOIs, it's easiest to submit the polygon directly instead of using the basemaps API to do the search for quad IDs.  E.g. you can submit an order similar to this:

```
   order = {
        "description": "foobar",
        "source_type": "basemaps",
        "order_type": "partial",
        "products": [
            {
                "mosaic_name": <mosaic_name>,
                "geometry": <direct_geojson_polygon>,
            }
        ],
        "tools": []
    }
```

However, that will fail if too many quads are in the polygon.  

The approach you have here of searching for the quads explicitly and then chunking orders up into N quads works around that and is a _very_ useful thing to demonstrate, as it's frequently needed.

It's a good thing to demonstrate, but I also wonder if it's too much complexity to show in a basic example?  I honestly don't know, though, and it's _definitely_ useful to show these sort of chunking approaches.  

At any rate, I defer to you on that. What you have currently is _great_! I think it's fine to merge as is. My comment was mostly me thinking out loud, and feel free to ignore it!

**Review by kevinlacaille — CHANGES_REQUESTED** [review]
Great improvement! 

I still need to take a look at why the last cell containing `await batch_lists_and_place_orders(quad_ids, polygon_search_order_params)` is rendering >20 lines between reports. I replaced the callback function with simply `print()` and it's still doing the same thing. I'll have to dive deeper into this. 

In the mean time, just check out my comments!

File: jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb [hunk 2]
```diff
@@ -434,29 +603,6384 @@
...
+    "\n",
+    "    \"\"\"\n",
+    "    base_url = 'https://api.planet.com/basemaps/v1'\n",
+    "\n",
+    "    if api_key is None:\n",
+    "        api_key = os.getenv('PL_API_KEY')\n",
+    "\n",
+    "    # Create a session and configure authentication\n",
+    "    session = requests.Session()\n",
+    "    session.auth = (api_key, '')\n",
```
[the remark this answers is not in the document]
Comment (kevinlacaille) [reply jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb:640]: Oh, and you can get rid of `api_key` in your inputs and docstring :)

**Review by kevinlacaille — CHANGES_REQUESTED** [review]
A few more little things!

File: jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb [hunk 3]
```diff
@@ -222,21 +234,38 @@
...
+    "from urllib3.util.retry import Retry"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 7,
+   "execution_count": 264,
    "metadata": {},
    "outputs": [],
    "source": [
```
[the remark this answers is not in the document]
Comment (kevinlacaille) [reply jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb:271]: If you do get rid of it, then you may remove `HTTPBasicAuth` from imports.

**Review by kevinlacaille — CHANGES_REQUESTED** [review]

File: jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb [hunk 4]
```diff
@@ -343,72 +420,167 @@
...
   {
    "cell_type": "code",
-   "execution_count": 10,
+   "execution_count": 269,
    "metadata": {},
    "outputs": [],
    "source": [
-    "def calculate_resolution(latitude):\n",
-    "    '''Calculate the resolution of a basemap at a given latitude\n",
+    "async def batch_lists_and_place_orders(quad_ids, order_params):\n",
```
Comment (kevinlacaille) [inline jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb:476]: The formatting looks weird on this function. Try as I noted in Slack, like this

```python
async def batch_lists_and_place_orders(quad_ids, order_params):
    """
    Process quad IDs in batches and create orders.

    Parameters:
        quad_ids (list): A list of quad IDs to be processed in batches.
        order_params (dict): The order parameters dictionary that contains the details of the order.

    """

    # Calculate the number of batches
    num_batches = math.ceil(len(quad_ids) / 100)

    # Create batched quad IDs lists
    batched_quad_ids = [
        quad_ids[i:i + 100] for i in range(0, len(quad_ids), 100)
    ]

    # Duplicate the order_params dictionary for each batch
    all_order_params = [
        copy.deepcopy(order_params) for _ in range(num_batches)
    ]

    # Assign batched quad IDs to each order_params dictionary
    for i, params in enumerate(all_order_params):
        params['products'][0]['quad_ids'] = batched_quad_ids[i]

    async with planet.Session() as ps:
        # The Orders API client
        client = ps.client('orders')

        # Create the order and deliver it to GCP for each batch
        await asyncio.gather(*[
            create_and_deliver_order(params, client)
            for params in all_order_params
        ])
```

File: jupyter-notebooks/orders_basemaps/requests_order_basemaps.ipynb [hunk 5]
```diff
@@ -518,57 +772,206 @@
...
+     "metadata": {},
+     "output_type": "execute_result"
     }
    ],
    "source": [
-    "r = requests.get(order_url, auth=session.auth)\n",
-    "response = r.json()\n",
-    "results = response['_links']['results']\n",
-    "print(\"\\n\".join([r['name'] for r in results]))"
+    "poll_for_success(order_url)"
```
Comment (kevinlacaille) [inline jupyter-notebooks/orders_basemaps/requests_order_basemaps.ipynb:974]: I think I'm confused what's supposed to be happening here. Why do we poll in this cell and the cell above? Do we need both?

File: jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb [hunk 6]
```diff
@@ -67,17 +67,21 @@
...
-   "execution_count": 1,
+   "execution_count": 257,
    "metadata": {},
    "outputs": [],
    "source": [
     "import planet\n",
-    "import os"
+    "import os\n",
+    "import copy\n",
+    "from urllib3.util.retry import Retry\n",
```
Comment (kevinlacaille) [inline jupyter-notebooks/orders_basemaps/SDK_order_basemaps.ipynb:77]: Get rid of either this one or the one below, please.

**Review by kevinlacaille — CHANGES_REQUESTED** [review]
A few more things.

Also, I can't comment on the function itself, but `poll_for_success()` no longer needs `num_loops` in its docstring.

**Review by kevinlacaille — APPROVED** [review]
This is a huge improvement, Emma! Great work 💪

**Pull request merged**